src/builder/yara_rule_builder.cpp

```diff
@@ -365,7 +365,7 @@ YaraRuleBuilder& YaraRuleBuilder::withHexString(const std::string& id, const std
 {
 	createLastString();
 
-	if (id == std::string{} || hexString->getText() == std::string{})
+	if (id.empty() || hexString->getText().empty())
```

Comment:
Consider adding a new member function to `hexString` called `empty()` or `is_empty()` to simplify the code and remove the [Law-of-Demeter violation](https://en.wikipedia.org/wiki/Law_of_Demeter) (more info in Czech is [here](https://cs-blog.petrzemek.net/2014-11-16-demeterin-zakon-aneb-minimalizujeme-strukturalni-zavislosti)).

Maybe I am taking things too far. I just wanted to provide an additional tip for consideration.